feature/google/src/test/java/com/revenuecat/purchases/google/FindBestOfferTest.kt

```diff
@@ -0,0 +1,529 @@
+        val days = parseBillPeriodToDays("P1D")
+        assertThat(days).isEqualTo(1)
+    }
+
+    @Test
+    fun `parsed period for P2D is 2 days`() {
+        val days = parseBillPeriodToDays("P2D")
+        assertThat(days).isEqualTo(2)
+    }
+
+    @Test
+    fun `default offer is base plan when no offers`() {
+        /*
+         * sub_1
+         *   |
+         *   |-> monthly (P1M, $4.99)
+         */
+
+        // Base plan
+        val monthlyBasePlan = mockSubscriptionOfferDetails(
+            offerId = "",
+            basePlanId = "monthly",
+            pricingPhases = listOf(mockPricingPhase(
+                price = 4.99,
+                billingPeriod = "P1M"
+            ))
+        )
+
+        // Product details
+        val productDetail1 = mockProductDetails(
+            productId = "sub_1",
+            type = BillingClient.ProductType.SUBS,
+            subscriptionOfferDetails = listOf(monthlyBasePlan))
+        val productDetails = listOf(productDetail1)
+
+        val storeProducts = productDetails.toStoreProducts()
+
+        val defaultOption = storeProducts.first().defaultOption as GoogleSubscriptionOption
+        assertThat(defaultOption.id).isEqualTo("monthly")
+    }
+
+    @Test
+    fun `default offer is only free offer`() {
+        /*
+         * sub_1
+         *   |
+         *   |-> monthly (P1M, $4.99)
+         *         |
+         *         |-> month-free (BEST OFFER)
+         *               |-> (P1M, free)
+         */
+
+        // Base plan
+        var monthlyBasePlanPricingPhase = mockPricingPhase(
+            price = 4.99,
+            billingPeriod = "P1M"
+        )
+        val monthlyBasePlan = mockSubscriptionOfferDetails(
+            offerId = "",
+            basePlanId = "monthly",
+            pricingPhases = listOf(monthlyBasePlanPricingPhase)
+        )
+
+        // month-free (P1M, free)
+        val monthFreeOffer = mockPricingPhase(
+            price = 0.0,
+            billingPeriod = "P1M",
+            billingCycleCount = 1,
+            recurrenceMode = ProductDetails.RecurrenceMode.NON_RECURRING
+        )
+        val freeOffer = mockSubscriptionOfferDetails(
+            offerId = "month-free",
+            basePlanId = "monthly",
+            pricingPhases = listOf(monthFreeOffer, monthlyBasePlanPricingPhase)
+        )
+
+        // Product details
+        val productDetail1 = mockProductDetails(
+            productId = "sub_1",
+            type = BillingClient.ProductType.SUBS,
+            subscriptionOfferDetails = listOf(monthlyBasePlan, freeOffer))
+        val productDetails = listOf(productDetail1)
+
+        val storeProducts = productDetails.toStoreProducts()
+
+        val defaultOption = storeProducts.first().defaultOption as GoogleSubscriptionOption
+        assertThat(defaultOption.id).isEqualTo("monthly:month-free")
+    }
+
+    @Test
+    fun `default offer is offer with longest free phase`() {
+        /*
+         * sub_1
+         *   |
```

Comment:
Minor: while i appreciate the comment explaining the scenario, it woula even be better if we could keep the code for setup this concise, so the comment isn't needed - see my prev. comment. Otherwise it means having to maintain this in case the test changes

Reply:
Test should be more readable now without the comments!